cpp/CMakeLists.txt

```diff
@@ -16,11 +16,8 @@
 
 cmake_minimum_required(VERSION 3.14...3.17 FATAL_ERROR)
 
-project(CUDA_DATAFRAME VERSION 0.19.0 LANGUAGES C CXX CUDA)
+project(CUDA_DATAFRAME VERSION 0.19.0 LANGUAGES C CXX)
```

Comment:
We don't need the CUDA language anymore?

Reply:
We still need the CUDA language enabled ( cpp/cmake/Modules/ConfigureCUDA.cmake:43 ) but we currently do it after we setup an initial value for `CMAKE_CUDA_ARCHITECTURES` ( cpp/cmake/Modules/SetGPUArchs.cmake:39 for reason ). 

Thinking about this more, I think we can improve the code so it is easier to follow by doing something like the following:
```cmake
cmake_minimum_required(VERSION 3.14...3.18 FATAL_ERROR)

# https://gitlab.kitware.com/cmake/cmake/-/issues/21302
if(NOT DEFINED CMAKE_CUDA_ARCHITECTURES)
  set(CUDF_BUILD_FOR_ALL_ARCHS TRUE)
elseif(CMAKE_CUDA_ARCHITECTURES STREQUAL "")
  set(CUDF_BUILD_FOR_HOST_ARCHS TRUE) 
endif()

project(CUDA_DATAFRAME VERSION 0.19.0 LANGUAGES C CXX CUDA)
...
include(SetGPUArchs.cmake) # now looks for CUDF_BUILD_FOR_ALL_ARCHS && CUDF_BUILD_FOR_HOST_ARCHS
```

This would allow cudf to obey `CMAKE_CUDA_ARCHITECTURES` values set by the user, but also allow `SetGPUArchs` to setup smart defaults to only build for the host machines hardware.

@kkraus14 @trxcllnt  Does this solution look reasonable to you?

Reply:
Yea that looks pretty clean to me. I think we have similar code in RMM as well.